src/consumer/worker.spec.js

```diff
@@ -37,4 +56,31 @@ describe('Worker', () => {
     await worker.run({ next })
     expect(reject).toHaveBeenCalledWith(error)
   })
+
+  it('should skip batches that are already assigned to another worker', async () => {
+    const secondWorker = createWorker({
+      handler,
+      workerId: 1,
+      partitionAssignments,
+      logger: newLogger(),
+    })
+
+    const secondNext = jest.fn(() => undefined)
+    const batches = seq(2).map(createBatch)
+
+    ;[next, secondNext].forEach(queue => {
```

Comment:
```suggestion
    [next, secondNext].forEach(queue => {
```

Reply:
This is there because otherwise the parser is going to think I'm trying to do `seq(2).map(createBatch)[next, secondNext].forEach(...`. I can make it a bit nicer by assigning the array to a variable before acting on it.

Reply:
oh right, don't need to change then.